src/Core/Settings.h

```diff
@@ -449,7 +449,6 @@ class IColumn;
     M(Bool, optimize_skip_merged_partitions, false, "Skip partitions with one part with level > 0 in optimize final", 0) \
     M(Bool, optimize_on_insert, true, "Do the same transformation for inserted block of data as if merge was done on this block.", 0) \
     M(Bool, allow_experimental_map_type, true, "Obsolete setting, does nothing.", 0) \
-    M(Bool, allow_experimental_window_functions, false, "Allow experimental window functions", 0) \
```

Comment:
Previously when I make map type not experimental I just converted setting to dummy
```
M(Bool, allow_experimental_map_type, true, "Obsolete setting, does nothing.", 0)
```
It seems that it can be just removed, right ?

Reply:
I think yes, because it was experimental. There might be some problems (not sure which problems exactly) when e.g. doing an upgrade, so we sometimes keep the old settings for several releases before removing them, but for experimental ones this is not needed.

Reply:
@kitaisreal This PR breaks everything:

```
2021.08.08 18:05:42.564335 [ 1433476 ] {} <Error> Application: DB::Exception: Unknown setting allow_experimental_window_functions
2021.08.08 18:05:42.564371 [ 1433476 ] {} <Information> Application: shutting down
2021.08.08 18:05:42.564377 [ 1433476 ] {} <Debug> Application: Uninitializing subsystem: Logging Subsystem
2021.08.08 18:05:42.564579 [ 1433477 ] {} <Trace> BaseDaemon: Received signal -2
2021.08.08 18:05:42.564616 [ 1433477 ] {} <Information> BaseDaemon: Stop SignalListener thread
2021.08.08 18:05:42.586497 [ 1433468 ] {} <Information> Application: Child process exited normally with code 70.
```